cocos/renderer/CCMaterial.cpp

```diff
@@ -442,6 +442,10 @@ Material* Material::clone() const
         for (const auto& technique: _techniques)
         {
             auto t = technique->clone();
+            t->_parent = material;
+            for (ssize_t i = 0; i < t->getPassCount(); i++) {
+                t->getPassByIndex(i)->_parent = t;
```

Comment:
I would prefer to move this `for loop` code into Technique::clone(), it seems to be better.